src/Polly.Core/Telemetry/ResilienceStrategyTelemetry.cs

```diff
@@ -19,7 +19,11 @@ internal ResilienceStrategyTelemetry(ResilienceTelemetrySource source, Telemetry
 
     internal TelemetryListener? Listener { get; }
 
-    internal ResilienceTelemetrySource TelemetrySource { get; }
+    /// <summary>
+    /// Gets BlahBlahBlah.
+    /// </summary>
+    /// <value>dummy.</value>
+    public ResilienceTelemetrySource TelemetrySource { get; } // Fix visibility
```

Comment:
If I add the following line to the `Polly.Core.csproj`:
```xml
<InternalsVisibleToProject Include="Polly.RateLimiting" />
```
then I don't have to change the access modifier from `internal` to `public`. 

But it causes lots of compile time error, like complaining about `RateLimiterResilienceStrategy`'s `ExecuteCore` should be `proctected internal` not just `proctected`, etc..

Do you have any other idea than making this property `public`?

Reply:
As far as I can see, we're currently doing this:

1. Exposing the source;
2. Using the source to pass to an exception;
3. Having the exception turn the context into a string;
4. Expose the string on the exception.

If that is the case, would this be better?

1. Add a method to the context to turn it into a string (i.e. have it own its own logic to do that so it's not far from the type and could misalign if we changed the context in the future and forgot we were stringifying it elsewhere);
2. Use that method to expose that string representation on the telemetry;
3. Pass that string to the exception;
4. Expose that string on the exception.

The above avoids the visibility problem, and makes the source string come from Polly.Core as the source-of-truth, rather than making something visible solely for rate limiting to make its own decisions about.

Reply:
With this approach we end up with following error:
> Type 'RateLimiterRejectedException' already defines a member called 'RateLimiterRejectedException' with the same parameter types

```
public RateLimiterRejectedException(string telemetrySource)
```

Reply:
So let's add a constructor that doesn't cause that problem then 😄 - for example, by manually specifying an exception message `new RateLimiterRejectedException("The source {blah} go boom", source)`.

I think it would be much neater to have to manually construct a message because we've already used `.ctor(string)` for an explicit exception than have to expose the source just to avoid that.

Reply:
Why not just expose this property? 

https://github.com/App-vNext/Polly/blob/ea15b525317d96af934ada7003b9ff3b6f6d52af/src/Polly.Core/Telemetry/ResilienceStrategyTelemetry.cs#L22

The `ResilienceStrategyTelemetry` is already public. I would indeed suggest to make the constructor of `ResilienceTelemetrySource` public, there is no reason for us to hinder the testability.

Reply:
@martintmk That was the original plan. But Martin suggested to do not expose that property rather have a method which returns a string representation of the `TelemetrySource`. Both of the approaches feel a bit clumsy so I don't have a preference.

What do you guys think? Which direction should we go?

Reply:
I played around with the idea on how we can uniformly introduce this to all exceptions:

https://github.com/App-vNext/Polly/pull/2352

wdyt?